### PR TITLE
Refactor game class

### PR DESCRIPTION
**Pull request by teodorciuraru**
# Summary
- Refactors the game classes to fit better in a MVC design.
- Every game is split in 3 objects: the model (`*Game.class`), the controller (`*Controller.class)`, the view (`_ViewController.class_).
- There is an controller, model and view from each games' MVC inherits. 
# Other changes
## `ToastHelper`

It now shows the Toast message over every other views, by iterating over the `RootViewController`'s presented views.

### CODE REVIEW

File: malaria-ios/ViewControllers/Info Hub and Games/Games Lists/Games/Rapid Fire/RapidFireGame.swift [hunk 1]
```diff
@@ -1,19 +1,12 @@
...
 
-class RapidFireGame: Game {
+// The Rapid Fire game model.
+
+class RapidFireGame {
   
   var entries: [RapidFireGameEntry] = []
-  
-  var achievementManager: RapidFireAchievementManager?
+  var _maximumScore: Int?
```
Comment (bphenriques) [inline malaria-ios/ViewControllers/Info Hub and Games/Games Lists/Games/Rapid Fire/RapidFireGame.swift]: Why optional?

File: malaria-ios/ViewControllers/Info Hub and Games/Games Lists/Games/Myth vs. Fact/MythVsFactViewController.swift [hunk 2]
```diff
@@ -121,9 +74,34 @@ class MythVsFactViewController: GameViewController {
...
+      _ in
+      self.yesView?.alpha = CGFloat(1.0)
+      self.noView?.alpha = CGFloat(1.0)
+      
+      self.view.userInteractionEnabled = true
+      
+      self.mythVsFactGameHandler.setNextStametent()
+    }
+    
+    if answer {
```
Comment (bphenriques) [inline malaria-ios/ViewControllers/Info Hub and Games/Games Lists/Games/Myth vs. Fact/MythVsFactViewController.swift]:
``` swift
let targetView = answer ? yesView : noView
targetView. (...)
```

File: malaria-ios/ViewControllers/Info Hub and Games/Games Lists/Games/Myth vs. Fact/InsetLabel.swift [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import UIKit
+
+/// A label that presents padding.
+
+class InsetLabel: UILabel {
```
Comment (bphenriques) [inline malaria-ios/ViewControllers/Info Hub and Games/Games Lists/Games/Myth vs. Fact/InsetLabel.swift:5]: Why not a UITextLabel inside a view?

File: malaria-ios/ViewControllers/Info Hub and Games/Games Lists/Games/GameViewController.swift [hunk 4]
```diff
@@ -56,66 +53,31 @@ class GameViewController: UIViewController {
...
+    {
       (action:UIAlertAction!) -> Void in
       
       self.dismissViewControllerAnimated(true, completion: nil)
     }
     
     alert.addAction(backAction)
     
-    let restartGameAction = UIAlertAction(title: "Restart", style: UIAlertActionStyle.Default) {
+    let restartGameAction = UIAlertAction(title: "Restart",
```
Comment (bphenriques) [inline malaria-ios/ViewControllers/Info Hub and Games/Games Lists/Games/GameViewController.swift]: Move this type of constans to a different extension like so:
https://github.com/systers/malaria-app-ios/blob/develop/malaria-ios/ViewControllers/PlanTrip/PlanTripViewController.swift#L313

Reason: Organized and easier to find/Replace/localize

File: malaria-ios/ViewControllers/Info Hub and Games/Games Lists/Games/Myth vs. Fact/InsetLabel.swift (hunk 3 above)
Comment (teodorciuraru) [inline malaria-ios/ViewControllers/Info Hub and Games/Games Lists/Games/Myth vs. Fact/InsetLabel.swift:5]: What can I respond? :D 

This is what I thought about. Is it wrong?

**Comment by bphenriques** [comment]
Resubmit after merging with the previous one.

**Pull request closed without merging**